dev-libs/softhsm/softhsm-2.3.0.recipe

```diff
@@ -0,0 +1,85 @@
+SUMMARY="An implementation of a cryptographic store"
+DESCRIPTION="SoftHSM is an implementation of a cryptographic store \
+accessible through a PKCS #11 interface. You can use it to explore PKCS #11 \
+without having a Hardware Security Module. It is being developed as a part \
+of the OpenDNSSEC project. SoftHSM uses Botan for its cryptographic \
+operations."
+HOMEPAGE="https://www.opendnssec.org/softhsm/"
+COPYRIGHT="2010 .SE, The Internet Infrastructure Foundation and SURFnet bv"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://dist.opendnssec.org/source/softhsm-$portVersion.tar.gz"
+CHECKSUM_SHA256="5ed604c89a3a6ef9d7d1ee92c28a2c4b3cd1f86f302c808e2d12c8f39aa2c127"
+SOURCE_DIR="softhsm-$portVersion"
+PATCHES="softhsm-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
+
+PROVIDES="
+	softhsm_2$secondaryArchSuffix = $portVersion
+	cmd:softhsm2_dump_file$commandSuffix = $portVersion
+	cmd:softhsm2_keyconv$commandSuffix = $portVersion
+	cmd:softhsm2_util$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	botan$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	botan${secondaryArchSuffix}_devel
+	cmd:awk
+	cmd:diff
+	cmd:find
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	cmd:pkg_config$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	"
+
+USER_SETTINGS_FILES="
+	settings/softhsm2.conf template data/softhsm/softhsm2.conf.sample
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/softhsm2.conf keep-old
+	var/lib/softhsm directory keep-old
```

Comment:
```diff
 GLOBAL_WRITABLE_FILES="
 	settings/softhsm2.conf keep-old
	var/lib/softhsm directory keep-old
 	"
+
+defineDebugInfoPackage softhsm2$secondaryArchSuffix \
+	$commandBinDir/softhsm2-dump-file \
+	$commandBinDir/softhsm2-keyconv \
+	$commandBinDir/softhsm2-util \
+	$libDir/libsofthsm2.so
+
 BUILD()
 {
```
I don't know if it's OK to have a `var/lib/softhsm` directory. We might need to change this in some way to make it adhere to the Haiku/HaikuPorts rules.

Reply:
Is the said directory included in the package? If yes then it's necessary that we include it, otherwise it have to be removed

From [BuildingPackages](https://dev.haiku-os.org/wiki/Obsolete/MovedToTree/PackageManagement/BuildingPackages)

> `global-writable-files`: A list of global writable file infos. The list elements must have the following format:
>
>      global_writable_file_info	::= path [ "directory" ] [ "keep-old" | "manual" | "auto-merge" ]
>
> `path` is the relative path of the writable file or directory, starting with "settings/" or any other writable directory. If the "directory" keyword is given, the path refers to a directory. If no other keyword is given after the path respectively after the "directory" keyword, the file or directory is not included in the package. It will be created by the software or by the user.

To be honest the definition of this variable implicts that we need one for every `USER_SETTINGS_FILES`. I'm a bit confused with this myself, probably needs some knowledgeable ppl to explain.